cookbook/add-tabs.rst

```diff
@@ -0,0 +1,289 @@
+                        ->setTabPriority(511)
+                        ->addToolbarActions($formToolbarActionsWithoutType)
+                        ->addRouterAttributesToFormRequest($routerAttributesToFormRequest)
+                        ->setPreviewCondition($previewCondition)
+                        ->setTitleVisible(true)
+                        ->setTabOrder(2048)
+                        ->setParent(PageAdmin::EDIT_FORM_VIEW)
+                );
+            }
+        }
+
+        private function hasSomeWebspacePermission(): bool
+        {
+            foreach ($this->webspaceManager->getWebspaceCollection()->getWebspaces() as $webspace) {
+                $hasWebspacePermission = $this->securityChecker->hasPermission(
+                    PageAdmin::SECURITY_CONTEXT_PREFIX . $webspace->getKey(),
+                    PermissionTypes::EDIT
+                );
+
+                if ($hasWebspacePermission) {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+    }
+
+We can register this class as a service and give it a tag of ``sulu.admin``, then it wil be picked up by Sulu.
+
+.. code-block:: yaml
+
+    app.social_admin:
+        class: App\Admin\PageAdminExtension
+        arguments:
+            - '@Sulu\Bundle\AdminBundle\Admin\View\ViewBuilderFactoryInterface'
+            - '@sulu_core.webspace.webspace_manager'
+            - '@sulu_security.security_checker'
+        tags:
+            - { name: 'sulu.admin'}
+            - { name: 'sulu.context', context: 'admin' }
+
+When you debug the container right now your should see your own ``Admin`` class show up.
+
+.. code-block:: bash
+
+    $ php bin/console debug:container --tag=sulu.admin
+
+        Service ID               Class name                                           
+        sulu_contact.admin       Sulu\Bundle\ContactBundle\Admin\ContactAdmin        
+        sulu_preview.admin       Sulu\Bundle\PreviewBundle\Admin\PreviewAdmin        
+        sulu_custom_urls.admin   Sulu\Bundle\CustomUrlBundle\Admin\CustomUrlAdmin    
+        sulu_website.admin       Sulu\Bundle\WebsiteBundle\Admin\WebsiteAdmin        
+        sulu_tag.admin           Sulu\Bundle\TagBundle\Admin\TagAdmin                
+        sulu_page.admin          Sulu\Bundle\PageBundle\Admin\PageAdmin              
+        sulu_snippet.admin       Sulu\Bundle\SnippetBundle\Admin\SnippetAdmin        
+        sulu_category.admin      Sulu\Bundle\CategoryBundle\Admin\CategoryAdmin      
+        sulu_security.admin      Sulu\Bundle\SecurityBundle\Admin\SecurityAdmin      
+        sulu_media.admin         Sulu\Bundle\MediaBundle\Admin\MediaAdmin            
+        sulu_search.admin        Sulu\Bundle\SearchBundle\Admin\SearchAdmin          
+        my_own.admin             App\Admin\PageAdminExtension  
+
+You should now see the tab show up but it's not working yet.
+
+
+Registering your structure extension 
```

Comment:
This is quite page specific. Should we use some more generic title like `Persist the data of the form`?
 @danrot